snuba/writer.py

```diff
@@ -58,24 +81,33 @@ def __default(self, value):
         else:
             raise TypeError
 
-    def __encode(self, row: WriterTableRow):
-        return json.dumps(row, default=self.__default).encode('utf-8')
+    def __encode(self, row: WriterTableRow) -> bytes:
+        return json.dumps(row, default=self.__default).encode("utf-8")
 
     def write(self, rows: Iterable[WriterTableRow]):
-        parameters = self.__options.copy()
-        parameters['query'] = f"INSERT INTO {self.__table_name} FORMAT JSONEachRow"
-        resp = self.__pool.urlopen(
-            'POST',
-            '/?' + urlencode(parameters),
-            headers={
-                'Connection': 'keep-alive',
-                'Accept-Encoding': 'gzip,deflate',
-            },
+        response = self.__pool.urlopen(
+            "POST",
+            "/?"
+            + urlencode(
+                {
+                    **self.__options,
+                    "query": f"INSERT INTO {self.__table_name} FORMAT JSONEachRow",
+                }
+            ),
+            headers={"Connection": "keep-alive", "Accept-Encoding": "gzip,deflate"},
             body=map(self.__encode, rows),
             chunked=True,
         )
-        if resp.status // 100 != 2:
-            raise HTTPError(f"{resp.status} Unexpected")
+
+        if response.status != 200:
```

Comment:
So this is different than what we did before. Do we expect anything other than 200 in the 2xx range ?

Reply:
I don't think so. The ClickHouse docs imply that everything is either a 200 or a 500, but I'm not sure I'd consider that a specification:

> If successful, you receive the 200 response code and the result in the response body. If an error occurs, you receive the 500 response code and an error description text in the response body.

- https://clickhouse.yandex/docs/en/interfaces/http/

Reply:
This doesn't directly address your question, but looks like a missing table actually is a 404, so I wouldn't consider that documentation line authoritative.

I still don't expect anything other than 200 on a successful write (e.g. no 202 status.)